app/src/main/java/io/horizontalsystems/bankwallet/core/storage/CoinTypeConverter.kt

```diff
@@ -0,0 +1,44 @@
+package io.horizontalsystems.bankwallet.core.storage
+
+import android.arch.persistence.room.TypeConverter
+import io.horizontalsystems.bankwallet.entities.CoinType
+
+
+class CoinTypeConverter {
+
+    private val bitcoinKey = "bitcoin_key"
+    private val bitcoinCashKey = "bitcoin_cash_key"
+    private val ethereumKey = "ethereum_key"
+    private val erc20Key = "erc_20_key"
+
+
+    @TypeConverter
+    fun stringToCoinType(value: String): CoinType {
+        return when(value) {
+            bitcoinKey -> CoinType.Bitcoin
+            bitcoinCashKey -> CoinType.BitcoinCash
+            ethereumKey -> CoinType.Ethereum
+            else -> {
+                if (value.contains(erc20Key)) {
+                    val parts = value.split(";")
+                    if (parts.size == 3) {
+                        CoinType.Erc20(parts[1], parts[2].toInt())
+                    }
+                }
+                throw Throwable()
```

Comment:
Let's use more detailed Exception instead of the general one. In addition, this part of code will always throw an exception, even if it is right ERC20.